pkg/binding/bindingpolicy-resolution.go

```diff
@@ -230,34 +253,52 @@ type objectRef struct {
 	cache.ObjectName
 }
 
-func bindingObjectRefToDataFromWorkload(bindingSpecWorkload *v1alpha1.DownsyncObjectClauses) map[objectRef]*objectData {
-	bindingObjectRefToData := make(map[objectRef]*objectData)
+func bindingObjectRefToDataFromWorkload(bindingSpecWorkload *v1alpha1.DownsyncObjectClauses) map[objectRef]*ObjectData {
+	bindingObjectRefToData := make(map[objectRef]*ObjectData)
 
 	for _, clusterScopeDownsyncClause := range bindingSpecWorkload.ClusterScope {
+		var statusCollectors sets.Set[string]
+		var stalenessThresholdSecs int32
+
+		if clusterScopeDownsyncClause.StatusCollection != nil {
+			statusCollectors = sets.New(clusterScopeDownsyncClause.StatusCollection.StatusCollectors...)
+			stalenessThresholdSecs = clusterScopeDownsyncClause.StatusCollection.StalenessThresholdSecs
+		}
+
 		bindingObjectRefToData[objectRef{
 			GroupVersionResource: schema.GroupVersionResource(clusterScopeDownsyncClause.GroupVersionResource),
 			ObjectName: cache.ObjectName{
 				Name:      clusterScopeDownsyncClause.Name,
 				Namespace: metav1.NamespaceNone,
 			},
-		}] = &objectData{
-			ResourceVersion:  clusterScopeDownsyncClause.ResourceVersion,
-			CreateOnly:       clusterScopeDownsyncClause.CreateOnly,
-			StatusCollectors: sets.New(clusterScopeDownsyncClause.StatusCollectors...),
+		}] = &ObjectData{
+			ResourceVersion:        clusterScopeDownsyncClause.ResourceVersion,
+			CreateOnly:             clusterScopeDownsyncClause.CreateOnly,
+			StatusCollectors:       statusCollectors,
+			StalenessThresholdSecs: stalenessThresholdSecs,
 		}
 	}
 
 	for _, namespacedScopeDownsyncClause := range bindingSpecWorkload.NamespaceScope {
+		var statusCollectors sets.Set[string]
+		var stalenessThresholdSecs int32
```

Comment:
Default is 120?

Reply:
Default is 120 when `StatusCollection` is none nil, which is true only when the object matched at least one StatusCollector.

If the latter is true, the threshold will be correctly set. Otherwise its value is useless.

Reply:
I think this is moot now, having realized that ManifestWork does not support the useful kind of "stale".